post-processor/vagrant/post-processor.go

```diff
@@ -83,11 +83,17 @@ func (p *PostProcessor) PostProcessProvider(name string, provider Provider, ui p
 
 	ui.Say(fmt.Sprintf("Creating Vagrant box for '%s' provider", name))
 
-	config.ctx.Data = &outputPathTemplate{
-		ArtifactId: artifact.Id(),
-		BuildName:  config.PackerBuildName,
-		Provider:   name,
+	generatedData := make(map[interface{}]interface{})
+	stateData := artifact.State("generated_data")
+	if stateData != nil {
+		// Make sure it's not a nil map so we can assign to it later.
+		generatedData = stateData.(map[interface{}]interface{})
```

Comment:
Watch out; this can still panic: https://play.golang.org/p/Ebk9T6wgwIa

This is a common pitfal of the go interface. An interface is made of a type and a value; so if the type is set and not the value; the map can still be nil.